mlflow/transformers/__init__.py

```diff
@@ -1183,7 +1175,23 @@ def _load_model(path: str, flavor_config, return_type: str, device=None, **kwarg
         conf["device_map"] = device_map_strategy
         accelerate_model_conf["device_map"] = device_map_strategy
         # Cannot use device with device_map
+        if device is not None:
+            raise MlflowException.invalid_parameter_value(
+                "The environment variable MLFLOW_HUGGINGFACE_USE_DEVICE_MAP is set to True, but "
+                f"the `device` argument is provided with value {device}. The device_map and "
+                "`device` argument cannot be used together. Set MLFLOW_HUGGINGFACE_USE_DEVICE_MAP "
+                "to False to specify a particular device ID, or pass None for the `device` "
+                "argument to use device_map."
+            )
         device = None
+    elif device is None:
+        if MLFLOW_DEFAULT_PREDICTION_DEVICE.get():
+            try:
+                device = int(MLFLOW_DEFAULT_PREDICTION_DEVICE.get())
+            except ValueError:
+                device = _TRANSFORMERS_DEFAULT_CPU_DEVICE_ID
```

Comment:
```suggestion
        if device_value := MLFLOW_DEFAULT_PREDICTION_DEVICE.get():
            try:
                device = int(device_value)
            except ValueError:
                _logger.warning(f"Invalid value for {MLFLOW_DEFAULT_PREDICTION_DEVICE}: {device_value!r}, setting device to {_TRANSFORMERS_DEFAULT_CPU_DEVICE_ID}")
                device = _TRANSFORMERS_DEFAULT_CPU_DEVICE_ID
```

can we insert a warning?

Reply:
Oops missed this, follow-up: https://github.com/mlflow/mlflow/pull/12991